### PR TITLE
solution of the 'json object processor' problem

### PR DESCRIPTION
**Pull request by koryun23**

### CODE REVIEW

File: src/main/com/aca/homework/week12/json/JsonObjectProcessor.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    private Object getValueOfField(Object o, Field field) {
+        try {
+            return field.get(o);
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+            throw new RuntimeException("Object " + o + " does not have field " + field);
+        }
+    }
+
+    private Field[] getFieldsOfObject(Object o){
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week12/json/JsonObjectProcessor.java:40]: the method name lies. it also mutates fields access

File: src/main/com/aca/homework/week12/json/JsonObjectProcessor.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    private String buildJsonObjectFromFields(Field[] fields, Object o){
+        StringBuilder sb = new StringBuilder();
+        sb.append("{");
+        for(Field field : fields) {
+            sb.append(field.getDeclaredAnnotation(JsonProperty.class).value());
+            sb.append(": ");
+            sb.append(getValueOfField(o, field));
+            sb.append(", ");
+        }
+        sb.delete(sb.length()-2, sb.length());
```
Comment (gevorgyanarman) [inline src/main/com/aca/homework/week12/json/JsonObjectProcessor.java:26]: please format the code

File: src/test/com/aca/homework/week12/json/JsonObjectProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    private JsonObjectProcessor jsonObjectProcessor;
+
+    @BeforeEach
+    public void setUp(){
+        jsonObjectProcessor = new JsonObjectProcessor();
+    }
+
+    @Test
+    public void testProcess(){
+        Assertions.assertEquals("{username: frosty_chebyshev, name: Anton, age: 25}",
```
Comment (gevorgyanarman) [inline src/test/com/aca/homework/week12/json/JsonObjectProcessorTest.java:17]: "s are missing

**Review by gevorgyanarman — CHANGES_REQUESTED** [review]

**Comment by koryun23** [comment]
done

**Pull request merged**